### ISSUE TITLE
use of undeclared crate or module `bytecheck` when deriving `ByteCheck`

### ISSUE THREAD

**Issue by mkatychev** [rkyv/rkyv#343]
I introduced an issue in this PR:
https://github.com/rkyv/rkyv/pull/340

the reexport for `rkyv::CheckBytes` is done on the `CheckBytes` trait rather than `bytecheck_derive::CheckBytes`:
https://github.com/rkyv/bytecheck/blob/512eee1cfcfef8eec43a334611cc4bb7e7db1775/bytecheck/src/lib.rs#L169

```rust
use rkyv::{Archive, Deserialize, Serialize};
// bytecheck can be used to validate your data if you want
use rkyv::CheckBytes;

#[derive(Archive, Deserialize, Serialize, Debug, PartialEq)]
// This will generate a PartialEq impl between our unarchived and archived types
#[archive(compare(PartialEq))]
// To use the safe API, you have to derive CheckBytes for the archived type
#[archive_attr(derive(rkyv::CheckBytes, Debug))]
struct Test {
    int: u8,
    string: String,
    option: Option<Vec<i32>>,
}

fn main() {
    let value = Test {
        int: 42,
        string: "hello world".to_string(),
        option: Some(vec![1, 2, 3, 4]),
    };

    // Serializing is as easy as a single function call
    let bytes = rkyv::to_bytes::<_, 256>(&value).unwrap();

    // Or you can customize your serialization for better performance
    // and compatibility with #![no_std] environments
    use rkyv::ser::{serializers::AllocSerializer, Serializer};

    let mut serializer = AllocSerializer::<0>::default();
    serializer.serialize_value(&value).unwrap();
    let bytes = serializer.into_serializer().into_inner();

    // You can use the safe API for fast zero-copy deserialization
    let archived = rkyv::check_archived_root::<Test>(&bytes[..]).unwrap();
    assert_eq!(archived, &value);

    // Or you can use the unsafe API for maximum performance
    let archived = unsafe { rkyv::archived_root::<Test>(&bytes[..]) };
    assert_eq!(archived, &value);

    // And you can always deserialize back to the original type
    let deserialized: Test = archived.deserialize(&mut rkyv::Infallible).unwrap();
    assert_eq!(deserialized, value);
}
```

the code snippet above produces the error below:

``` --> src/main.rs:9:23
  |
9 | #[archive_attr(derive(CheckBytes, Debug))]
  |                       ^^^^^^^^^^ use of undeclared crate or module `bytecheck`
  |
  = note: this error originates in the derive macro `CheckBytes` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by mkatychev** [comment]
Serde manages to reexport both `serde_derive::Serialize` and `serde::ser::Serialize` in the top level module:
https://github.com/serde-rs/serde/blob/f85c4f2fa99c7f3b103e6e2580e75eb9313b00d0/serde/src/lib.rs#L321-L332

Trying to see if that can be reproduced since replacing `pub use bytecheck::CheckBytes;` with `pub use bytecheck_derive::CheckBytes;` complains that the trait is missing

**Comment by djkoloski** [comment]
This is the same as https://github.com/serde-rs/serde/issues/1465#issuecomment-800686252, which uses a custom attribute to set the path to the crate. Bytecheck could support this but currently does not. It's a proc-macro modification.

**Comment by mkatychev** [comment]
Awesome, I'll look into it.

**Comment by samlich** [comment]
Should be addressed by #344
